modules/server/src/main/scala/scaladex/server/service/UserSessionSynchronizer.scala

```diff
@@ -0,0 +1,65 @@
+package scaladex.server.service
+
+import java.util.UUID
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.StatusCodes
+import cats.implicits.toTraverseOps
+import scaladex.core.model.GithubResponse
+import scaladex.core.model.UserInfo
+import scaladex.core.model.UserState
+import scaladex.core.service.GithubService
+import scaladex.core.service.SchedulerDatabase
+import scaladex.infra.GithubClient
+
+class UserSessionSynchronizer(database: SchedulerDatabase)(implicit ec: ExecutionContext, ac: ActorSystem)
+    extends Scheduler("user-session-synchronizer", 1.hour) {
+
+  type Session = (UUID, UserState)
+
+  override def run(): Future[Unit] =
+    for {
+      sessions <- database.getAllSessions()
+      updatedEitherUserSessions <- sessions.traverse { case (session, client) => updateUserSession(session, client) }
+      expiredSessionIds = updatedEitherUserSessions.collect { case Left(expiredSessionId) => expiredSessionId }
+      updatedUserSessions = updatedEitherUserSessions.collect { case Right(sessions) => sessions }
+      _ <- updatedUserSessions.traverse { case (userId, userState) => database.insertSession(userId, userState) }
+      _ <- expiredSessionIds.traverse(database.deleteSession)
```

Comment:
I would suggest to do the `database.insertSession` or `database.deleteSession` inside the `updateUserSession` method, because we don't need to wait for all the user sessions to start inserting (or deleting) them in the database. Also I think this would make the code a bit more concise.